crypto/merkle.go

```diff
@@ -124,10 +124,12 @@ func MerkleProof(b []byte, proofIndex uint64) (base []byte, hashSet []Hash) {
 
 	// Get the proof and convert it to a base + hash set.
 	_, proof, _, _ := t.Prove()
-	base = proof[0]
-	hashSet = make([]Hash, len(proof)-1)
-	for i, p := range proof[1:] {
-		copy(hashSet[i][:], p)
+	if len(proof) > 0 {
```

Comment:
is it possible for `proof` to have less than one element? If it only happens when `len(b) == 0`, perhaps it would be better to check for that case explicitly.
Even if not, I think edge cases should be handled "off the main path," i.e.
```go
if len(proof) == 0 {
    return nil, nil
}
```